metricbeat/mb/testing/testdata.go

```diff
@@ -350,12 +365,19 @@ func documentedFieldCheck(foundKeys common.MapStr, knownKeys map[string]interfac
 			if found {
 				continue
 			}
-
-			// last case `status_codes.*`:
+			// case `status_codes.*`:
 			prefix := strings.Join(splits[0:len(splits)-1], ".")
 			if _, ok := knownKeys[prefix+".*"]; ok {
 				continue
 			}
+			// should cover scenarios as status_codes.*.*` and `azure.compute_vm_scaleset.*.*`
+			if len(splits) > 2 {
+				prefix = strings.Join(splits[0:len(splits)-2], ".")
+				if _, ok := knownKeys[prefix+".*.*"]; ok {
+					continue
+				}
+			}
+
```

Comment:
WDYT about adding unit tests for this function? I see it has started to grow and I foresee we will keep updating it for a while

Reply:
well spotted, added a test in `testdata_test.go`